### PR TITLE
Redo Source Table Steps

### LINKED ISSUE #421: [FEATURE] Upload photometry to SkyPortal
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I perform a reduction and the results just sit on my computer.

**Describe the solution you'd like**
I'd like for the photometry results to be uploaded to Fritz!

### LINKED ISSUE #448: [FEATURE] WINTER pipeline end-to-end
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when the WINTER data is not fully processed.

**Describe the solution you'd like**
WINTER data is fully processed.

### LINKED ISSUE #524: [FEATURE] End-to-end image subtraction
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I realise that many of our problems regarding image subtraction come from these features not being tested properly.

Specifically #463 , #464 and others result because we are testing image subtraction on non-mirar reduced images, separate from our mirar reduction tests.

A variety of other things, such as nans breaking scorr image generation, all stem from a lack of end-to-end tests.

**Describe the solution you'd like**
All pipelines that do imsub should have end-to-end tests. We should replace the summer_imsub/wirc_imsub tests with full reduction + subtraction.

Will be partially addressed by #448

### LINKED ISSUE #593: [FEATURE] Refactor send_to_fritz.py
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I just want to upload photometry to Fritz, but I have to also upload a cutout. I'm also always confused when the naming conventions in send_to_fritz.py (and other processors) are "candidate" and "alert" when my instrument of choice is used for follow-up. I also wonder if there is some duplicate code between `IPACAvroExporter` and `SendToFritz`. I'm also always frustrated when there is a "FRITZ_AUTHID" key when "FRITZ_TOKEN" could suffice.

**Describe the solution you'd like**
I'd like there to be an option like `SendToFritz.alert_put_photometry()` that only uploads photometry. I'd also like for it to be clear that the `alert` argument throughout send_to_fritz.py can just be a `SourceTable` object.

### LINKED ISSUE #602: [FEATURE] "Send to Skyportal" into two processors
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I realise send to Skyportal is overly complicated. 

**Describe the solution you'd like**
Having the "push candidate" functionality is useful, so we should not delete, but a new processor which just uploads saves a source and uploads photometry is probably enough. The new processor will be used by SEDMv2 (see #593 ) and WINTER.

### LINKED ISSUE #605: [BUG] Source writing should not require subtraction
**Describe the bug**
https://github.com/winter-telescope/mirar/blob/5a524701042a63d47ca45e3f7d142043de65b013/mirar/processors/sources/source_exporter.py#L67

### LINKED ISSUE #609: [FEATURE] Deduplicate Photometry
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when we have completely parallel photometry code for images and source tables.

**Describe the solution you'd like**
We already have code to extract a source at a particular position. I think all pipelines should basically do "reduce image", "optionally extract", "extract source(s)", "process sources". The latter category, including pushing to skyportal etc, should all be source processors not image processors. I think that we can cover every science case in this scheme, but I think we should discuss it to be sure.

### PR DESCRIPTION
**Pull request by robertdstein**
This PR reworks use of source tables to ensure consistency:

* It removes the duplication of photometry processors (fix #609). Now there are only source table photometry options. It renames several processors to ensure this change. It moves pipelines to use forced photometry source generation where needed.
* It removes a lot of unneeded copying of entries from metadata to source table data. Now, the table itself is only for things specific to detection, like RA/DEC while image-level properties (e.g zeropoint) stay in the metadata. Processors can access both metadata and source table rows.
* It ensures proper source table savings, which now regardless of whether image subtraction has been run (Fix #605).
* It substantially refactors skyportal code  (fix #593). It separates uploading candidates from uploading photometry (fix #602)
* It fixes many many hardcoded bugs in the send_to_fritz processor (Fix #421). For example, photometry is not always sent as J-band data now!
* Reorganises code a bit (progress on #198).
* Remove winter imsub test, introduces a single proper end-to-end winter test (fix #524 and fix #448).
* Ensures every prerequisite check raises the proper prerequisite error, and then ensures that if pipeline visualisation fails due to a prerequisite error, the correct pipeline and config is printed for easier debugging.

### CODE REVIEW

**Review by virajkaram — CHANGES_REQUESTED** [review]

File: mirar/pipelines/winter/winter_pipeline.py [hunk 1]
```diff
@@ -46,19 +50,27 @@ class WINTERPipeline(Pipeline):
...
+        "test": load_test
+        + csvlog
+        + extract_all
+        + mask_and_split
+        + select_split_subset
+        + save_raw
+        + full_reduction
+        + imsub
+        + detect_candidates
+        + process_candidates,
```
[the remark this answers is not in the document]
Comment (robertdstein) [reply mirar/pipelines/winter/winter_pipeline.py:67]: Get rid of test? That's the thing that the is run by the test winter script.

File: mirar/processors/astromatic/swarp/swarp.py [hunk 2]
```diff
@@ -416,7 +416,12 @@ def _apply_to_images(
 
         new_image["COADDS"] = sum(x["COADDS"] for x in batch)
         new_image[EXPTIME_KEY] = sum(x[EXPTIME_KEY] for x in batch)
-        new_image[TIME_KEY] = min(x[TIME_KEY] for x in batch)
+
+        for key in ["JD", "MJD", "DATE-OBS", "DATE-END", "MJD-OBS", TIME_KEY]:
```
[the remark this answers is not in the document]
Comment (robertdstein) [reply mirar/processors/astromatic/swarp/swarp.py:420]: Well I'm open to suggestions. The reason it's needed is our avro schema, and skyportal integration require JD. So we need to assign JD somewhere. Where would you prefer that to happen? Could also do it in those processors I guess.

File: mirar/pipelines/winter/winter_pipeline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (virajkaram) [reply mirar/pipelines/winter/winter_pipeline.py:67]: I meant the test_imsub

File: mirar/processors/astromatic/swarp/swarp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (virajkaram) [reply mirar/processors/astromatic/swarp/swarp.py:420]: Yeah, I would maybe just make sure TIME_KEY is transferred throughout processors and gets transformed to required format in the processor.

**Pull request merged**